### PR TITLE
[Lambda Cloud] Update docs and add `[lambda]` install option

### PR DESCRIPTION
**Pull request by ewzeng**
Add Lambda Cloud to documentation and add the `[lambda]` option. (The `[lambda]` option does not install anything extra, but is included for uniformity).

### CODE REVIEW

File: docs/source/getting-started/installation.rst [hunk 1]
```diff
@@ -22,7 +22,9 @@ only have access to certain clouds, use any combination of
 :code:`"[aws,azure,gcp]"` (e.g., :code:`"[aws,gcp]"`) to reduce the
 dependencies installed.
 
-You may also install SkyPilot from source.
+SkyPilot also supports a fourth cloud provider, `Lambda <https://lambdalabs.com/>`_, if you install SkyPilot from source.
+
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:26]: We plan to cut a release soon - so perhaps we can remove these changes?

File: docs/source/getting-started/installation.rst [hunk 2]
```diff
@@ -22,7 +22,9 @@ only have access to certain clouds, use any combination of
 :code:`"[aws,azure,gcp]"` (e.g., :code:`"[aws,gcp]"`) to reduce the
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:22]: With the release planned, maybe lets add lambda here and after L16.
```suggestion
:code:`"[aws,azure,gcp, lambda]"` (e.g., :code:`"[aws,gcp]"`) to reduce the
```

File: docs/source/getting-started/installation.rst [hunk 3]
```diff
@@ -92,6 +94,20 @@ Note: if you encounter *Authorization Error (Error 400: invalid_request)* with t
 
 Hint: run ``az account subscription list`` to get a list of subscription IDs under your account.
 
+**Lambda**
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:97]: Lambda or Lambda Cloud?
```suggestion
**Lambda Cloud**
```

File: docs/source/getting-started/installation.rst [hunk 4]
```diff
@@ -92,6 +94,20 @@ Note: if you encounter *Authorization Error (Error 400: invalid_request)* with t
...
+
+.. code-block:: text
+
+  api_key = your_api_key_here
+
+to :code:`~/.lambda_cloud/lambda_keys`.
+
+.. note::
+
+  Currently, SkyPilot does not support multi-node clusters or stopping instances on Lambda Cloud.
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:109]:
```suggestion
  Multi-node clusters and stopping instances is currently not supported on Lambda Cloud.
```

File: docs/source/index.rst [hunk 5]
```diff
@@ -25,7 +25,7 @@ SkyPilot is a framework for easily and cost effectively running ML workloads on
 
 SkyPilot abstracts away cloud infra burden:
 
-- Launch jobs & clusters on any cloud (AWS, Azure, GCP)
+- Launch jobs & clusters on any cloud (AWS, Azure, GCP, Lambda)
```
Comment (romilbhardwaj) [inline docs/source/index.rst:28]: Let's update this line in README.md too!

File: docs/source/getting-started/installation.rst [hunk 6]
```diff
@@ -92,6 +94,20 @@ Note: if you encounter *Authorization Error (Error 400: invalid_request)* with t
...
 
+**Lambda**
+
+Create the file :code:`~/.lambda_cloud/lambda_keys`, go to `API Keys <https://cloud.lambdalabs.com/api-keys>`_ to generate a key, and then add the line
+
+.. code-block:: text
+
+  api_key = your_api_key_here
+
+to :code:`~/.lambda_cloud/lambda_keys`.
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:105]: How about this? Good to give direct instructions to users. (Please test if this actually works first :) )
 
```suggestion
Go to `API Keys <https://cloud.lambdalabs.com/api-keys>`_ to generate a key, and then add it to `~/.lambda_cloud/lambda_keys` by running:

.. code-block:: console
  $ # Create directories if required
  $ mkdir -p ~/.lambda_cloud
  $ # Add api_key = <your_api_key_here> line to lambda_keys file
  $ echo "api_key = <your_api_key_here>" > ~/.lambda_cloud/lambda_keys
```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @ewzeng! Left some comments. Haven't tried running any tests yet.

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:97]: I put Lambda because we refer it to it as 'Lambda' on the first page ('Lambda Cloud' felt too wordy). But a longer name might do better here.

File: docs/source/index.rst (hunk 5 above)
Comment (ewzeng) [reply docs/source/index.rst:28]: Good point! I totally forgot about that.

File: docs/source/getting-started/installation.rst (hunk 6 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:105]: Nice! And I will test.

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst:22]: nit: no extra space

File: docs/source/getting-started/installation.rst (hunk 4 above)
Comment (concretevitamin) [reply docs/source/getting-started/installation.rst:109]: nit: is -> are?

File: docs/source/index.rst (hunk 5 above)
Comment (concretevitamin) [reply docs/source/index.rst:28]: nit: Lambda Cloud?

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:97]: Actually, since it seems @concretevitamin prefers Lambda Cloud over Lambda, I'll change everything to Lambda Cloud.

File: docs/source/getting-started/installation.rst [hunk 7]
```diff
@@ -92,6 +94,20 @@ Note: if you encounter *Authorization Error (Error 400: invalid_request)* with t
 
 Hint: run ``az account subscription list`` to get a list of subscription IDs under your account.
 
+**Lambda**
+
+Create the file :code:`~/.lambda_cloud/lambda_keys`, go to `API Keys <https://cloud.lambdalabs.com/api-keys>`_ to generate a key, and then add the line
```
Comment (concretevitamin) [inline docs/source/getting-started/installation.rst:99]: If this is to become the first mention of Lambda, maybe have one sentence "Lambda Cloud <with link> is a cloud provider offering low-cost GPUs." since not everyone may know it.

Comment (ewzeng) [reply docs/source/getting-started/installation.rst:99]: Good point! I will add.

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:22]: Sounds good!

File: docs/source/getting-started/installation.rst (hunk 1 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:26]: Agreed

Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated!

File: docs/source/getting-started/installation.rst (hunk 2 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated!

File: docs/source/getting-started/installation.rst (hunk 3 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated!

File: docs/source/index.rst (hunk 5 above)
Comment (ewzeng) [reply docs/source/index.rst]: Updated!

File: docs/source/getting-started/installation.rst (hunk 4 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated (with is -> are).

File: docs/source/getting-started/installation.rst (hunk 6 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated (and tested!)

File: docs/source/getting-started/installation.rst (hunk 7 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: I added a short description with a link to the Lambda Labs website

**Comment by ewzeng** [comment]
Thanks for the comments @romilbhardwaj @concretevitamin!

File: docs/source/getting-started/installation.rst [hunk 8]
```diff
@@ -92,6 +93,23 @@ Note: if you encounter *Authorization Error (Error 400: invalid_request)* with t
 
 Hint: run ``az account subscription list`` to get a list of subscription IDs under your account.
 
+**Lambda Cloud**
+
+Lambda Labs GPU Cloud is a cloud provider offering low-cost GPUs. You can learn more about them `here <https://lambdalabs.com/>`__.
+
+To configure Lambda Cloud access, go to `API Keys <https://cloud.lambdalabs.com/api-keys>`_ to generate a key and then add it to :code:`~/.lambda_cloud/lambda_keys` by running:
```
Comment (romilbhardwaj) [inline docs/source/getting-started/installation.rst:100]: Nit - make the location of the API Keys page clear for folks who can't see/access hyperlink
```suggestion
To configure Lambda Cloud access, go to `API Keys <https://cloud.lambdalabs.com/api-keys>`_ page on your Lambda Cloud console to generate a key and then add it to :code:`~/.lambda_cloud/lambda_keys` by running:
```

**Review by romilbhardwaj — APPROVED** [review]
LGTM - thanks @ewzeng!

File: docs/source/getting-started/installation.rst (hunk 8 above)
Comment (ewzeng) [reply docs/source/getting-started/installation.rst:100]: Good idea.

Comment (ewzeng) [reply docs/source/getting-started/installation.rst]: Updated!
